migration/transform/base.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+
+
+class Transform(ABC):
+    """Base class for data transformation."""
+
+    @abstractmethod
+    def run(self):
+        """Transform and yield one element at a time."""
+        pass
+
+class RDMRecordTransform(ABC):
+    """Base class for data transformation."""
+
+    @abstractmethod
+    def _id(self, entry):
+        """Transform the id."""
+        pass
+
+    @abstractmethod
+    def _record(self, entry):
+        """Transform the record."""
+        pass
+
+    @abstractmethod
+    def _draft(self, entry):
```

Comment:
not 100% convinced, this assumes that from one entry we can extract all that information. When in reality this comes because we are transforming Zenodo records (which have `_deposit`). However, it makes sense to have record, draft, parent and files as a unit...